plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,590 @@
+    export_status = export_license(self, module, payload, export_license_url)
+    return export_status
+
+
+def export_license_cifs(self, module):
+    uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    export_license_url = f"{uri[0]}/{EXPORT_LICENCE_NETWORK_SHARE_URI}"
+    payload = {}
+    payload["EntitlementID"] = module.params.get('license_id')
+    payload["ShareType"] = "CIFS"
+    if module.params.get('share_parameters').get('workgroup'):
+        payload["Workgroup"] = module.params.get('share_parameters').get('workgroup')
+    share_details = get_share_details(module)
+    payload.update(share_details)
+    export_status = export_license(self, module, payload, export_license_url)
+    return export_status
+
+
+def export_license_nfs(self, module):
+    uri = validate_and_get_first_resource_id_uri(
+        self.module, self.idrac, SYSTEMS_URI)
+    export_license_url = f"{uri[0]}/{EXPORT_LICENCE_NETWORK_SHARE_URI}"
+    payload = {}
+    payload["EntitlementID"] = module.params.get('license_id')
+    payload["ShareType"] = "NFS"
+    payload["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+    payload["ShareName"] = module.params.get('share_parameters').get('share_name')
+    export_status = export_license(self, module, payload, export_license_url)
+    return export_status
+
+
+def export_license(self, module, payload, export_license_url):
+    license_name = module.params.get('share_parameters').get('file_name')
+    if license_name:
+        license_file_name = f"{license_name}_iDRAC_license.xml"
+    else:
+        license_file_name = f"{module.params['license_id']}_iDRAC_license.xml"
+    payload["FileName"] = license_file_name
+    license_status = self.idrac.invoke_request(export_license_url, "POST", data=payload)
+    return license_status
+
+
+def get_share_details(module):
+    share_details = {}
+    share_details["IPAddress"] = module.params.get('share_parameters').get('ip_address')
+    share_details["ShareName"] = module.params.get('share_parameters').get('share_name')
+    share_details["UserName"] = module.params.get('share_parameters').get('username')
+    share_details["Password"] = module.params.get('share_parameters').get('password')
+    return share_details
+
+
+def main():
+    specs = get_argument_spec()
+    specs.update(idrac_auth_params)
+    module = AnsibleModule(
+        argument_spec=specs,
+        mutually_exclusive=[
+            ("import", "export", "delete")],
+        required_if=[
+            ["import", True, ("share_parameters",)],
+            ["export", True, ("license_id", "share_parameters",)],
+            ["delete", True, ("license_id",)]
+        ],
+        supports_check_mode=False)
+
+    try:
+        with iDRACRedfishAPI(module.params) as idrac:
+            idrac_firmware_version = get_idrac_firmware_version(idrac)
+            if LooseVersion(idrac_firmware_version) <= '3.0':
```

Comment:
When we use dynamic URI's the version check is not required, show not supported when the URI dosn't exist.

Reply:
No changes were made. Using current approach